docs/source-pytorch/clouds/cluster_intermediate_2.rst

```diff
@@ -35,3 +35,13 @@ Run the below command with the appropriate variables set on each node.
 .. note::
 
     ``torch.distributed.run`` assumes that you'd like to spawn a process per GPU if GPU devices are found on the node. This can be adjusted with ``-nproc_per_node``.
+
+.. note::
+
+    This approach will not work if attempted to be used in a SLURM environment. To make it work you need to set at the top of your main script:
+
+.. code-block:: python
+
+    import os
+
+    os.environ["SLURM_JOB_NAME"] = "interactive"
```

Comment:
Thanks for bringing this up @stas00. I agree we should make this more obvious for users who want to run torchrun on their slurm instances. My only suggestion here is to keep the note more concise and point to the SLURM document as well:

```suggestion
.. note::

    If you are running on a SLURM cluster, you need to explicitly enable interactive mode by setting the job name or adding ``os.environ["SLURM_JOB_NAME"] = "interactive"`` to the beginning of your script. 
```


Furthermore, I'd like to propose we make an explicit warning in Lightning to inform the user when they run on SLURM but have launched with torchelastic and suggesting the same action as you did here (can be done in a separate PR). What do you think?

Reply:
I think this should be fixed in PL not to require neither warning nor hacks to work. You have multiple env vars set by `torchrun` so it should be trivial to not activate SLURMEnvironment if these are set.

```
TORCHELASTIC_ERROR_FILE=/tmp/torchelastic__xw_o4n8/none_5gasx5xq/attempt_0/0/error.json
TORCHELASTIC_MAX_RESTARTS=0
TORCHELASTIC_RESTART_COUNT=0
TORCHELASTIC_RUN_ID=none
TORCHELASTIC_USE_AGENT_STORE=False
```

Reply:
Also I tend to repeat this very often when someone proposes to add a warning when an assert is needed - how will the warning help here? if you know ahead of time that SLURM and torchrun can't work together then detect that situation and assert telling the user how they can fix it.

Warnings don't work in modern ML software since it emits tens of thousands of warnings and there is no way users can even introspect the warnings. 

So my personal opinion is that warnings should be abolished altogether, there should be just 2 actions - things work - then move along, things don't work - assert.

Reply:
We are fully aligned that we would prefer to make a change in the code rather than adding warnings in the docs. That's why I made the proposal - because I think users won't easily make that step to go to the docs first when something goes wrong. You are absolutely right, we should investigate whether the conflict for detection can be resolved reliably. I'll check if I can make the detection this way without breaking any tests. 
But I don't think your message had to be so negative and frustrating to make a point. Here at Lightning we are very open to change and will make the change when necessary, history should show this. This is the first time I heard of your frustration regarding the SLURM environment and I'm very sorry. If you ever have a suggestion for improvement, we are always happy and responsive if you open a GH issue or shoot us a message in our channels.

Reply:
oh, my sincere apologies, @awaelchli - I can totally see how my message came across as dumping all the blame of the world on you. It wasn't the intention, I was just trying to signal that a better solution is needed than a warning.

I have spent many hours yesterday trying to figure out why my launching worked in an interactive slurm mode and would fail in a normal sbatch mode. So I was just hoping a solution will be found that will prevent many other users go through a painful figuring out of why the 2 ways the slurm job was running weren't doing the same thing. And using a warning will not be it.

Mea culpa for bad communications!

Reply:
Also, I don't yet know the nuances of SLURMEnvironment - but perhaps the other solution is that it could be made aware of torchelastic and simply not try to allocate yet another port?

Reply:
No worries it's all good. It's just that so far nobody has brought up this use case that you had. Normally users would rely on SLURM to create all processes and then Lightning detects that and off we go. In #18618 I'm going to try to enable both the standard use case and yours. 

SLURM_JOB_NAME=interactive is just a way to tell Lightning that you intend to use the machine in SLURM like a regular machine, and that Lightning shouldn't bother about any slurm settings. SLURMEnvironment is only important if the SLURM cluster creates the processes and Lightning needs to behave accordingly.

Reply:
Thank you for accepting my apologies and clarifying the situation - perhaps adding some of your notes to the docs will go a long way helping new users find their way around faster?